pkg/build/controller/factory/factory.go

```diff
@@ -323,13 +323,35 @@ type podLW struct {
 
 // List lists all Pods that have a build label.
 func (lw *podLW) List() (runtime.Object, error) {
-	sel, _ := labels.Parse(buildapi.BuildLabel)
-	return lw.client.Pods(kapi.NamespaceAll).List(sel, fields.Everything())
+	// get builds with new label
+	sel, err := labels.Parse(buildapi.BuildLabel)
+	if err != nil {
+		return nil, err
+	}
+	listNew, err := lw.client.Pods(kapi.NamespaceAll).List(sel, fields.Everything())
+	if err != nil {
+		return nil, err
+	}
+	// FIXME: get builds with old label - remove this when depracated label will be removed
+	selOld, err := labels.Parse(buildapi.DeprecatedBuildLabel)
+	if err != nil {
+		return nil, err
+	}
+	listOld, err := lw.client.Pods(kapi.NamespaceAll).List(selOld, fields.Everything())
+	if err != nil {
+		return nil, err
+	}
+	listNew.Items = append(listNew.Items, listOld.Items...)
```

Comment:
Don't assume we won't run a migration or that won't happen - always defend
against that.

On Jul 7, 2015, at 7:23 AM, Maciej Szulik <notifications@github.com> wrote:

In pkg/build/controller/factory/factory.go
<https://github.com/openshift/origin/pull/3508#discussion_r34029858>:

> +		return nil, err
> +	}
> +	listNew, err := lw.client.Pods(kapi.NamespaceAll).List(sel, fields.Everything())
> +	if err != nil {
> +		return nil, err
> +	}
> +	// FIXME: get builds with old label - remove this when depracated label will be removed
> +	selOld, err := labels.Parse(buildapi.DeprecatedBuildLabel)
> +	if err != nil {
> +		return nil, err
> +	}
> +	listOld, err := lw.client.Pods(kapi.NamespaceAll).List(selOld, fields.Everything())
> +	if err != nil {
> +		return nil, err
> +	}
> +	listNew.Items = append(listNew.Items, listOld.Items...)

Should not happen, since old pods will have old label and new - the new
one. The only time when the two would be there is a human intervention to
add the second label.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3508/files#r34029858>.